### PR TITLE
feat: initial bot code

### PR DESCRIPTION
**Pull request by nhcarrigan**
#### Proposed Changes

This PR establishes some initial bot code. The changes include:
- A bot configuration object (to allow passing values between commands, such as the prefix)
- A defined interface for command files
- Manual compiling of commands into an array
- A `help` and `ping` command, for testing purposes

After merge, still to do:
- Implement error handling for failed commands, login, etc.
- Migrate the manual command list to an automatic function

Steps to test these changes:
- Pull down changes locally
- Add your bot token to the `.env` (visit the [Discord Developer Portal]() to create an application if you do not have one)
- Install dependencies with `npm ci`
- Build the TypeScript files with `npm run build`
- Start the application with `npm start`
  - Confirm you see "It's aliiiiiive" in the terminal - this means the bot logged in
- Visit a Discord channel where your bot application is present
  - Send the message "!stargate ping" and see the bot's response
  - Send the message "!stargate help" and see the bot's response
  - Send the message "!stargate help help" and see the bot's response
  - Send the message "!stargate hahaha" and see no response from the bot

### CODE REVIEW

File: src/interfaces/CommandInt.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import { Message } from "discord.js";
+import { BotConfigInt } from "./BotConfigInt";
+
+export interface ComamndInt {
```
Comment (eddiejaoude) [inline src/interfaces/CommandInt.ts:4]: I think I might better to append with the full word `Interface` instead of the abbreviation `Int`

Comment (nhcarrigan) [reply src/interfaces/CommandInt.ts]: Fixed in commit `eca1b2a`.

File: src/commands/general/ping.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import { CommandInterface } from "../../interfaces/CommandInterface";
+
+export const ping: CommandInterface = {
```
Comment (Vyvy-vi) [inline src/commands/general/ping.ts:3]: lgtm, ~~but can we also have a pong command~~ (for documentation purpose, ofcourse)

**Review by Vyvy-vi — DISMISSED** [review]
LGTM-!

File: src/commands/general/ping.ts (hunk 2 above)
Comment (nhcarrigan) [reply src/commands/general/ping.ts:3]: What would it do and why?

Comment (Vyvy-vi) [reply src/commands/general/ping.ts:3]: nvm 😅  (ref: symmetric documentation, hf)

**Review by AllanRegush — COMMENTED** [review]
Looks good to me 👍 Maybe we should add a link from the `readme`  to `creating-commands.md`.

**Comment by nhcarrigan** [comment]
Great idea @AllanRegush - let's create that as an issue after this merges so we can open it up to first timers.

**Review by eddiejaoude — DISMISSED** [review]
Looks good 👍

**Review by eddiejaoude — APPROVED** [review]
LGTM

**Pull request merged**